kubectl-plugin/pkg/util/generation/generation.go

```diff
@@ -0,0 +1,175 @@
+package generation
+
+import (
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	resource "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+)
+
+type RayClusterYamlObject struct {
+	ClusterName    string
+	Namespace      string
+	RayVersion     string
+	Image          string
+	HeadCPU        string
+	HeadMemory     string
+	WorkerGrpName  string
+	WorkerCPU      string
+	WorkerMemory   string
+	WorkerReplicas int32
+}
+
+// Uncessary fields or error causing fields within the RayClusters that get filled in when generating the yaml
+var fieldsToRemove = []string{
```

Comment:
Do we still need this?

Reply:
If so, why do we need it?

Reply:
We do need this with the current approach. It gives an error when applying the yaml if the fields are filled in. The time stamp fields and the status fields aren't supposed to be filled in with any value.

Reply:
>  It gives an error when applying the yaml if the fields are filled in.

But the YAML is generated by the CLI, so it shouldn't be setting any uncessary fields right?

Reply:
Discussed offline. Currently with the typing of `creationTimestamp` as a Time struct, it causes the evaluation to ignore the `omitempty` and populate the field with null values. We will manually remove these fields until when/if `creationTimestamp` is changed to a pointer.